### PR TITLE
fix #7 added support for private/protected fields

### LINKED ISSUE #7: Allow serialization/deserialization of private fields
Currently the library enforces models to expose all their fields as public - custom fields as well as _id etc. This is considered bad practice in OOP. Both Moshi and GSON can use `field.setAccessible(true)` to implement this feature - so should moshi-jsonapi.

### PR DESCRIPTION
**Pull request by mreichelt**
- also added unit tests for serialization/deserialization
- also covered the case where fields might be inherited from parent classes
- also added TestResources to load real .json files (which are easier to read than Java strings with all the quotes)

### CODE REVIEW

**Comment by mreichelt** [comment]
@joachimr please review

File: src/test/java/moe/banana/jsonapi2/model/Private.java [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+import moe.banana.jsonapi2.JsonApi;
+import moe.banana.jsonapi2.Resource;
+
+@JsonApi(type = "private")
+public class Private extends Resource {
+
+    private String a;
+    private Double b;
+    private Integer c;
+    private Boolean d;
```
Comment (JoachimR) [inline src/test/java/moe/banana/jsonapi2/model/Private.java]: maybe name the values aString, aDouble, anInteger, aBoolean ?

File: src/test/java/moe/banana/jsonapi2/model/Protected.java [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+import moe.banana.jsonapi2.JsonApi;
+import moe.banana.jsonapi2.Resource;
+
+@JsonApi(type = "private")
+public class Protected extends Resource {
+
+    protected String a;
+    protected Double b;
+    protected Integer c;
+    protected Boolean d;
```
Comment (JoachimR) [inline src/test/java/moe/banana/jsonapi2/model/Protected.java]: maybe rename the values someString, someDouble, someInteger, someBoolean ?

**Comment by JoachimR** [comment]
review done

**Comment by mreichelt** [comment]
@kamikat I added support for private & protected fields. Hope you like it. :)
My colleague @joachimr already gave it a review, too.

**Comment by kamikat** [comment]
Merged. And version 2.1.0 released.

**Pull request merged**